src/bin/migrate.rs

```diff
@@ -0,0 +1,127 @@
+//! This binary is used to migrate user IDs from base64 to hex encoding.
+use std::{path::Path, str::FromStr};
+
+use anyhow::{anyhow, Context, Result};
+use base64::{engine::general_purpose, Engine as _};
+use famedly_sync::Config;
+use futures::StreamExt;
+use tracing::level_filters::LevelFilter;
+use zitadel_rust_client::v2::{
+	users::{
+		ListUsersRequest, SearchQuery, SetHumanProfile, TypeQuery, UpdateHumanUserRequest,
+		UserFieldName, Userv2Type,
+	},
+	Zitadel,
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+	// Config
+	let config_path =
+		std::env::var("FAMEDLY_SYNC_CONFIG").unwrap_or_else(|_| "./config.yaml".to_owned());
+	let config = Config::new(Path::new(&config_path))?;
+
+	// Tracing
+	let subscriber = tracing_subscriber::FmtSubscriber::builder()
+		.with_max_level(
+			config
+				.log_level
+				.as_ref()
+				.map_or(Ok(LevelFilter::INFO), |s| LevelFilter::from_str(s))?,
+		)
+		.finish();
+	tracing::subscriber::set_global_default(subscriber)
+		.context("Setting default tracing subscriber failed")?;
+
+	tracing::info!("Starting migration");
+	tracing::debug!("Old external IDs will be base64 decoded and re-encoded as hex");
+	tracing::debug!("Note: External IDs are stored in the nick_name field of the user's profile in Zitadel, often referred to as uid.");
+
+	// Zitadel
+	let zitadel_config = config.zitadel.clone();
+	let mut zitadel = Zitadel::new(zitadel_config.url, zitadel_config.key_file).await?;
+
+	// Get all users
+	let mut stream = zitadel.list_users(
```

Comment:
Should we reuse the method from the ldap source for this? Then we don't have to do the user conversion here as well.

Reply:
Which method do you mean? `get_next_zitadel_user`?

Reply:
This method: https://github.com/famedly/famedly-sync/blob/main/src/zitadel.rs#L91

Reply:
Took a bit of refactoring, but here we go.